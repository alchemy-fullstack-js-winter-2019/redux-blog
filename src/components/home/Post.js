import React from 'react';
import PropTypes from 'prop-types';

export default function Post({ post }) {
  return (
    <>
      <h2>{post.title}</h2>
      <p>{post.body}</p>
    </>
  );
}

Post.propTypes = {
  post: PropTypes.object.isRequired
};
