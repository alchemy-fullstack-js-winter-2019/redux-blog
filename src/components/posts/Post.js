import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles1 from '../app/App.css';

export default function Post({ post }) {
  return (
    <>
      <Link to={`/posts/${post.id}`} key={post.id} className={styles1.link}><h3>{post.title}</h3></Link>
      <p className={styles1.postBody}>{post.body}</p>
    </>
  );
}

Post.propTypes = {
  post: PropTypes.object.isRequired
};
