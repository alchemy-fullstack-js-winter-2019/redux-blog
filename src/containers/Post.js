import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PostDetail from '../components/posts/PostDetail';
import { getPost } from '../selectors/posts/posts';
import { fetchPost } from '../action/posts';
import { withRouter } from 'react-router-dom';
import AllComments from './AllComments';
class Post extends PureComponent {
  static propTypes = {
    post: PropTypes.object.isRequired,
    fetch: PropTypes.func.isRequired,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string.isRequired
      })
    })
  };

  componentDidMount() {
    this.props.fetch(this.props.match.params.id);
  }

  render() {
    return (
      <>
        <PostDetail post={this.props.post}/>
        <h2>Comments:</h2>
        <AllComments />
      </>
    );
  }
}

const mapStateToProps = state => ({
  post: getPost(state)
});

const mapDispatchToProps = (dispatch, props) => ({
  fetch() {
    dispatch(fetchPost(props.match.params.id));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(Post));
