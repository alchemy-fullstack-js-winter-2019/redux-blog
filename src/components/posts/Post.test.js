import React from 'react';
import renderer from 'react-test-renderer';
import Post from './Post';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Post', () => {
  it('matches a snapshot', () => {
    const post = {
      title: 'title',
      body: 'body'
    };
    const tree = renderer.create(
      <Router>
        <Post post={post}/>
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
