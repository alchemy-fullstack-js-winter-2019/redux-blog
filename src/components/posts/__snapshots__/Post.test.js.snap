// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches a snapshot 1`] = `
Array [
  <a
    href="/posts/undefined"
    onClick={[Function]}
  >
    <h2>
      title
    </h2>
  </a>,
  <p>
    body
  </p>,
]
`;
